src/core/nonbonded_interactions/nonbonded_interaction_data.hpp

```diff
@@ -414,15 +415,15 @@ class InteractionsNonBonded {
    * @return Reference to interaction parameters for the type pair.
    */
   auto &get_ia_param(int i, int j) {
-    return *m_nonbonded_ia_params[get_ia_param_key(i, j)];
+    return *m_nonbonded_ia_params.at(get_ia_param_key(i, j));
```

Comment:
Only in `Debug` builds. In `Release` builds, if you make a mistake in the checkpointing mechanism and reload the wrong value for `max_seen_particle_type`, you silently enter undefined behavior territory. The `at()` version sends a signal even in `Release` builds with `-DNDEBUG`.

Reply:
On second thought, this is actually redundant, because the default-constructed `shared_ptr` will be a null pointer, and not a default-constructed non-bonded interaction. Thus, any mistake will be immediately visible in a `Release` build. Also, I just found out this check can't be disabled with compiler flag `-U_GLIBCXX_ASSERTIONS`.